### PR TITLE
with: init at 0.0.1

### PR DESCRIPTION
**Pull request by acowley**
###### Motivation for this change

Adds the "with" shell helper application.
###### Things done
- [ ] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [ ] NixOS
  - [x] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/with-shell/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{ stdenv, fetchFromGitHub }:
+stdenv.mkDerivation {
+  name = "with";
```
Comment (rasendubi) [inline pkgs/applications/misc/with-shell/default.nix]: Add version to `name`.
```nix
name = "with-0.0.1";
```

`version` field is usually used as a variable to reuse later and is not used by `mkDerivation` itself, so you may remove it.

File: pkgs/applications/misc/with-shell/default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    rev = "cc2828bddd92297147d4365765f4ef36385f050a";
+    sha256 = "10m2xv6icrdp6lfprw3a9hsrzb3bip19ipkbmscap0niddqgcl9b";
+  };
+  installPhase = ''
+    mkdir -p $out/bin
+    cp with $out/bin/with
+  '';
+  meta = {
+    homepage = "https://github.com/mchav/With";
+    description = "Command prefixing for continuous workflow using a single tool.";
```
Comment (rasendubi) [inline pkgs/applications/misc/with-shell/default.nix]: Remove dot at end of `description`.

File: pkgs/applications/misc/with-shell/default.nix (hunk 1 above)
Comment (rasendubi) [inline pkgs/applications/misc/with-shell/default.nix]: If a package is install from git and does not have a version, you better use the date of the commit as version:
```
name = "with-2016-08-20";
```

**Comment by acowley** [comment]
Thank you for the comments @rasendubi !

**Comment by rasendubi** [comment]
Commit/PR can now be renamed to `with: init at 2016-08-20`

**Pull request merged**